examples/job/work-queue-2/README.md

```diff
@@ -0,0 +1,282 @@
+
+
+Here is an overview of the steps in this example:
+
+1. **Start a storage service to hold the work queue.**  In this example, we use Redis to store
+  our work items.  In the previous example, we used RabbitM.  In this example, we use Redis and
+  a custom work-queue client library because AMQP does not provide a good way for clients to
+  detect when a finite-length work queue is empty.  In practice you would set up a store such
+  as Redis once and reuse it for the work queues of many jobs, and other things.
+1. **Create a queue, and fill it with messages.**  Each message represents one task to be done.  In
+  this example, a message is just an integer that we will do a lengthy computation on.
+1. **Start a Job that works on tasks from the queue**.  The Job starts several pods.  Each pod takes
+  one task from the message queue, processes it, and repeats until the end of the queue is reached.
+
+
+## Starting Redis
+
+For this example, for simplicitly, we will start a single instance of Redis.
+See the [Redis Example](../../../examples/redis/README.md) for an example
+of deploying Redis scaleably and redundantly.
+
+Start a temporary Pod running Redis and a service so we can find it.
+
+```console
+$ kubectl create -f examples/job/work-queue-2/redis-pod.yaml
+pod "redis-master" created
+$ kubectl create -f examples/job/work-queue-2/redis-service.yaml
+service "redis" created
+```
+
+## Filling the Queue with tasks
+
+Now lets fill the queue with some "tasks".  In our example, our tasks are just strings to be
+printed.
+
+Start a temporary interactive pod for running the Redis CLI
+
+```console
+$ kubectl run -i --tty temp --image redis --command "/bin/sh"
+Waiting for pod default/redis2-c7h78 to be running, status is Pending, pod ready: false
+Hit enter for command prompt
+```
+
+Now hit enter, start the redis CLI, and create a list with some work items in it.
+
+```
+# redis-cli -h redis
+redis:6379> rpush job2 "apple"
+(integer) 1
+redis:6379> rpush job2 "banana"
+(integer) 2
+redis:6379> rpush job2 "cherry"
+(integer) 3
+redis:6379> rpush job2 "date"
+(integer) 4
+redis:6379> rpush job2 "fig"
+(integer) 5
+redis:6379> rpush job2 "grape"
+(integer) 6
+redis:6379> rpush job2 "lemon"
+(integer) 7
+redis:6379> rpush job2 "melon"
+(integer) 8
+redis:6379> rpush job2 "orange"
+(integer) 9
+redis:6379> lrange job2 0 -1
+1) "apple"
+2) "banana"
+3) "cherry"
+4) "date"
+5) "fig"
+6) "grape"
+7) "lemon"
+8) "melon"
+9) "orange"
+```
+
+So, the list with key `job2` will be our work queue.
+
+
+
+## Create an Image
+
+Now we are ready to create an image that we will run.
+
+We will use a python worker program with a redis client to read
+the messages from the message queue.
+
+The Redis work queue client is [provided](rediswq.py).
+
+
+
+```console
+$ mkdir tmp
+$ cd tmp
+$ cat > worker.py <<EOF
+#! /usr/bin/python
+import time
+import rediswq
+
+q = rediswq.RedisWQ(name="job2", host="redis")
```

Comment:
Since the hostname resolution did not work for me I had to use env var here again `os.getenv("REDIS_SERVICE_HOST")`